lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -541,28 +560,102 @@ def _handle_changed_secrets(self, event) -> None:
             event.defer()
             return
 
+        # FUTURE PR: if config-server does not have TLS enabled log a useful message and go into
+        # blocked in relation_changed and other status checks
+
+    def get_membership_auth_modes(self, event: RelationChangedEvent) -> Tuple[bool, bool]:
+        """Returns the available authentication membership forms."""
+        key_file_contents = self.database_requires.fetch_relation_field(
+            event.relation.id, KEYFILE_KEY
+        )
+        tls_ca = self.database_requires.fetch_relation_field(event.relation.id, INT_TLS_CA_KEY)
+        return (key_file_contents is not None, tls_ca is not None)
+
+    def update_member_auth(
+        self, event: RelationChangedEvent, membership_auth: Tuple[bool, bool]
+    ) -> None:
+        """Updates the shard to have the same membership auth as the config-server."""
+        cluster_auth_keyfile, cluster_auth_tls = membership_auth
```

Comment:
feel free to ignore: Is there any reason why a Tuple is needed as input if we convert it to two variables later?

Reply:
I wanted to be consistent with it's "sister" function `get_membership_auth_modes` which returns a `Tuple`